src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/RefreshableRegistryEntry.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot.registry;
+
+import java.time.Instant;
+
+public interface RefreshableRegistryEntry {
```

Comment:
Similar comment here. You can likely get this behavior without forcing the registries to implement some entry interface. Imagine the cache itself tracking this information separate from the object itself.